[WARN ][2014-04-28 13:08:40,443][com.mchange.v2.c3p0.impl.DefaultConnectionTester] - SQL State '08001' of Exception which occurred during a Connection test (fallback DatabaseMetaData test) implies that the database is invalid, and the pool should refill itself with fresh Connections.
com.ibm.db2.jcc.am.DisconnectNonTransientConnectionException: [jcc][t4][2030][11211][3.64.96] 在对连接的底层 socket、socket input stream 或 socket output stream 执行操作期间
发生通信错误。错误位置：Reply.fill() - socketInputStream.read (-1)。消息：Read timed out。 ERRORCODE=-4499, SQLSTATE=08001
	at com.ibm.db2.jcc.am.bd.a(bd.java:321)
	at com.ibm.db2.jcc.t4.a.a(a.java:548)
	at com.ibm.db2.jcc.t4.a.a(a.java:543)
	at com.ibm.db2.jcc.t4.x.b(x.java:252)
	at com.ibm.db2.jcc.t4.x.c(x.java:360)
	at com.ibm.db2.jcc.t4.x.c(x.java:473)
	at com.ibm.db2.jcc.t4.x.v(x.java:1241)
	at com.ibm.db2.jcc.t4.ab.f(ab.java:97)
	at com.ibm.db2.jcc.t4.p.e(p.java:81)
	at com.ibm.db2.jcc.t4.qb.k(qb.java:160)
	at com.ibm.db2.jcc.am.io.jb(io.java:2322)
	at com.ibm.db2.jcc.am.jo.b(jo.java:4401)
	at com.ibm.db2.jcc.am.CallableStatement.ec(CallableStatement.java:148)
	at com.ibm.db2.jcc.am.DatabaseMetaData.executeCatalogQuery(DatabaseMetaData.java:7813)
	at com.ibm.db2.jcc.am.DatabaseMetaData.getTablesX(DatabaseMetaData.java:6325)
	at com.ibm.db2.jcc.am.DatabaseMetaData.getTables(DatabaseMetaData.java:6231)
	at com.mchange.v2.c3p0.impl.DefaultConnectionTester.activeCheckConnectionNoQuery(DefaultConnectionTester.java:185)
	at com.mchange.v2.c3p0.impl.DefaultConnectionTester.activeCheckConnection(DefaultConnectionTester.java:62)
	at com.mchange.v2.c3p0.AbstractConnectionTester.activeCheckConnection(AbstractConnectionTester.java:67)
	at com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool$1PooledConnectionResourcePoolManager.testPooledConnection(C3P0PooledConnectionPool.java:368)
	at com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool$1PooledConnectionResourcePoolManager.refurbishIdleResource(C3P0PooledConnectionPool.java:310)
	at com.mchange.v2.resourcepool.BasicResourcePool$AsyncTestIdleResourceTask.run(BasicResourcePool.java:1999)
	at com.mchange.v2.async.ThreadPoolAsynchronousRunner$PoolThread.run(ThreadPoolAsynchronousRunner.java:547)
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.read(SocketInputStream.java:129)
	at com.ibm.db2.jcc.t4.x.b(x.java:208)
	... 19 more
[WARN ][2014-04-28 13:08:41,054][com.mchange.v2.c3p0.impl.DefaultConnectionTester] - SQL State '08001' of Exception which occurred during a Connection test (fallback DatabaseMetaData test) implies that the database is invalid, and the pool should refill itself with fresh Connections.
com.ibm.db2.jcc.am.DisconnectNonTransientConnectionException: [jcc][t4][2030][11211][3.64.96] 在对连接的底层 socket、socket input stream 或 socket output stream 执行操作期间
发生通信错误。错误位置：Reply.fill() - socketInputStream.read (-1)。消息：Read timed out。 ERRORCODE=-4499, SQLSTATE=08001
	at com.ibm.db2.jcc.am.bd.a(bd.java:321)
	at com.ibm.db2.jcc.t4.a.a(a.java:548)
	at com.ibm.db2.jcc.t4.a.a(a.java:543)
	at com.ibm.db2.jcc.t4.x.b(x.java:252)
	at com.ibm.db2.jcc.t4.x.c(x.java:360)
	at com.ibm.db2.jcc.t4.x.c(x.java:473)
	at com.ibm.db2.jcc.t4.x.v(x.java:1241)
	at com.ibm.db2.jcc.t4.ab.f(ab.java:97)
	at com.ibm.db2.jcc.t4.p.e(p.java:81)
	at com.ibm.db2.jcc.t4.qb.k(qb.java:160)
	at com.ibm.db2.jcc.am.io.jb(io.java:2322)
	at com.ibm.db2.jcc.am.jo.b(jo.java:4401)
	at com.ibm.db2.jcc.am.CallableStatement.ec(CallableStatement.java:148)
	at com.ibm.db2.jcc.am.DatabaseMetaData.executeCatalogQuery(DatabaseMetaData.java:7813)
	at com.ibm.db2.jcc.am.DatabaseMetaData.getTablesX(DatabaseMetaData.java:6325)
	at com.ibm.db2.jcc.am.DatabaseMetaData.getTables(DatabaseMetaData.java:6231)
	at com.mchange.v2.c3p0.impl.DefaultConnectionTester.activeCheckConnectionNoQuery(DefaultConnectionTester.java:185)
	at com.mchange.v2.c3p0.impl.DefaultConnectionTester.activeCheckConnection(DefaultConnectionTester.java:62)
	at com.mchange.v2.c3p0.AbstractConnectionTester.activeCheckConnection(AbstractConnectionTester.java:67)
	at com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool$1PooledConnectionResourcePoolManager.testPooledConnection(C3P0PooledConnectionPool.java:368)
	at com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool$1PooledConnectionResourcePoolManager.refurbishIdleResource(C3P0PooledConnectionPool.java:310)
	at com.mchange.v2.resourcepool.BasicResourcePool$AsyncTestIdleResourceTask.run(BasicResourcePool.java:1999)
	at com.mchange.v2.async.ThreadPoolAsynchronousRunner$PoolThread.run(ThreadPoolAsynchronousRunner.java:547)
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.read(SocketInputStream.java:129)
	at com.ibm.db2.jcc.t4.x.b(x.java:208)
	... 19 more
[WARN ][2014-04-28 13:08:41,164][com.mchange.v2.c3p0.impl.DefaultConnectionTester] - SQL State '08001' of Exception which occurred during a Connection test (fallback DatabaseMetaData test) implies that the database is invalid, and the pool should refill itself with fresh Connections.
com.ibm.db2.jcc.am.DisconnectNonTransientConnectionException: [jcc][t4][2030][11211][3.64.96] 在对连接的底层 socket、socket input stream 或 socket output stream 执行操作期间
发生通信错误。错误位置：Reply.fill() - socketInputStream.read (-1)。消息：Read timed out。 ERRORCODE=-4499, SQLSTATE=08001
	at com.ibm.db2.jcc.am.bd.a(bd.java:321)
	at com.ibm.db2.jcc.t4.a.a(a.java:548)
	at com.ibm.db2.jcc.t4.a.a(a.java:543)
	at com.ibm.db2.jcc.t4.x.b(x.java:252)
	at com.ibm.db2.jcc.t4.x.c(x.java:360)
	at com.ibm.db2.jcc.t4.x.c(x.java:473)
	at com.ibm.db2.jcc.t4.x.v(x.java:1241)
	at com.ibm.db2.jcc.t4.ab.f(ab.java:97)
	at com.ibm.db2.jcc.t4.p.e(p.java:81)
	at com.ibm.db2.jcc.t4.qb.k(qb.java:160)
	at com.ibm.db2.jcc.am.io.jb(io.java:2322)
	at com.ibm.db2.jcc.am.jo.b(jo.java:4401)
	at com.ibm.db2.jcc.am.CallableStatement.ec(CallableStatement.java:148)
	at com.ibm.db2.jcc.am.DatabaseMetaData.executeCatalogQuery(DatabaseMetaData.java:7813)
	at com.ibm.db2.jcc.am.DatabaseMetaData.getTablesX(DatabaseMetaData.java:6325)
	at com.ibm.db2.jcc.am.DatabaseMetaData.getTables(DatabaseMetaData.java:6231)
	at com.mchange.v2.c3p0.impl.DefaultConnectionTester.activeCheckConnectionNoQuery(DefaultConnectionTester.java:185)
	at com.mchange.v2.c3p0.impl.DefaultConnectionTester.activeCheckConnection(DefaultConnectionTester.java:62)
	at com.mchange.v2.c3p0.AbstractConnectionTester.activeCheckConnection(AbstractConnectionTester.java:67)
	at com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool$1PooledConnectionResourcePoolManager.testPooledConnection(C3P0PooledConnectionPool.java:368)
	at com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool$1PooledConnectionResourcePoolManager.refurbishIdleResource(C3P0PooledConnectionPool.java:310)
	at com.mchange.v2.resourcepool.BasicResourcePool$AsyncTestIdleResourceTask.run(BasicResourcePool.java:1999)
	at com.mchange.v2.async.ThreadPoolAsynchronousRunner$PoolThread.run(ThreadPoolAsynchronousRunner.java:547)
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.read(SocketInputStream.java:129)
	at com.ibm.db2.jcc.t4.x.b(x.java:208)
	... 19 more
[INFO ][2014-04-28 13:15:37,901][com.cherrypicks.boc.dao.impl.SystemUserRoleMapDaoImpl] - result size:1 sql:select r.* from BOC_CAMPAIGN.SYSTEM_ROLE r, BOC_CAMPAIGN.SYSTEM_USER_ROLE_MAP map, BOC_CAMPAIGN.SYSTEM_USER u where map.ROLE_ID = r.ID and map.USER_ID = u.ID and u.USER_NAME = 'admin'
[INFO ][2014-04-28 13:15:38,220][com.cherrypicks.boc.dao.impl.SystemRoleFunctionMapDaoImpl] - result size:5 sql:select sf.* from BOC_CAMPAIGN.SYSTEM_FUNCTION sf, BOC_CAMPAIGN.SYSTEM_ROLE_FUNCTION_MAP rfm where sf.ID = rfm.FUNC_ID and rfm.ROLE_ID in (:roleIds) 
